micro-rdk/src/common/webrtc/api.rs

```diff
@@ -299,47 +300,92 @@ impl WebRtcSignalingChannel {
-                status: Some(Status {
-                    code: Code::ResourceExhausted.into(),
-                    message: "too many active connections".to_string(),
-                    ..Default::default()
-                }),
-            })),
-        };
-
-        if let Err(e) = self.signaling_tx.send_message(answer).await {
-            log::error!("error sending signaling message: {:?}", e);
-            Err(WebRtcError::SignalingDisconnected())
-        } else {
-            log::warn!("too many active connections");
-            Ok(())
+        match self.signaling.as_mut() {
+            Either::Left(ref mut app_signaling) => {
+                let answer = AnswerResponse {
+                    uuid,
+                    stage: Some(answer_response::Stage::Error(AnswerResponseErrorStage {
+                        status: Some(Status {
+                            code: Code::ResourceExhausted.into(),
+                            message: "too many active connections".to_string(),
+                            ..Default::default()
+                        }),
+                    })),
+                };
+
+                if let Err(e) = app_signaling.tx.send_message(answer).await {
+                    log::error!("error sending signaling message: {:?}", e);
+                    Err(WebRtcError::SignalingDisconnected())
+                } else {
+                    log::warn!("too many active connections");
+                    Ok(())
+                }
+            }
+            Either::Right(ref mut local_signaling) => {
+                if let Err(e) = local_signaling
+                    .tx
+                    .send(Err(GrpcError::RpcResourceExhausted.into()))
+                    .await
+                {
+                    log::error!("error sending signaling message: {:?}", e);
+                    Err(WebRtcError::SignalingDisconnected())
+                } else {
+                    log::warn!("too many active connections");
+                    Ok(())
+                }
+            }
         }
     }
 
     pub(crate) async fn send_sdp_answer(&mut self, sdp: &WebRtcSdp) -> Result<(), WebRtcError> {
-        let answer = SdpOffer {
-            sdp_type: "answer".to_owned(),
-            sdp: sdp.sdp.marshal(),
-        };
-        let answer = self
-            .engine
-            .encode(serde_json::to_string(&answer).map_err(WebRtcError::AnswerMarshalError)?);
-
-        let answer = AnswerResponse {
-            uuid: sdp.uuid.clone(),
-            stage: Some(answer_response::Stage::Init(AnswerResponseInitStage {
-                sdp: answer,
-            })),
-        };
-        match self.signaling_tx.send_message(answer).await {
-            Err(e) => {
-                log::error!("error sending signaling message: {:?}", e);
-                Err(WebRtcError::SignalingDisconnected())
+        match self.signaling.as_mut() {
```

Comment:
could some of this be simplified by a function like `fn send_answer_stage(&mut self, uuid: String, stage: AnswerStage) -> Result<(), WebRtcError>`?